docs/cicd.md

```diff
@@ -0,0 +1,194 @@
+- [Environment variables](#environment-variables-and-dependencies)
+## Workflows
+
+Several workflows exist for different purposes:
+
+| Workflow                                   | Schedule/Trigger | Checks              | native delivery | backend_administration delivery | Version bump       |
+|--------------------------------------------|------------------|---------------------|-----------------|---------------------------------|--------------------|
+| commit                                     | commits of PRs   | :heavy_check_mark:  | :x:             | :x:                             | :x:                |
+| commit_main                                | commits on main  | :heavy_check_mark:  | :x:             | :x:                             | :x:                |
+| delivery_beta_all                          | manual           | :heavy_check_mark:  | beta            | beta                            | :heavy_check_mark: |
+| promote_all                                | manual           | :x:                 | promotion       | promotion                       | :x:                |
+| delivery_beta_backend_administration       | manual           | :heavy_check_mark:  | :x:             | beta                            | :heavy_check_mark: |
+| promote_backend_administration             | manual           | :x:                 | :x:             | promotion                       | :x:                |
+| delivery_beta_native                       | manual           | :heavy_check_mark:  | beta            | :x:                             | :heavy_check_mark: |
+| promote_native                             | manual           | :x:                 | production      | :x:                             | :x:                |
+| delivery_native_production                 | manual           | :heavy_check_mark:  | production      | :x:                             | :heavy_check_mark: |
+| delivery_production_backend_administration | manual           | :heavy_check_mark:  | :x:             | production                      | :heavy_check_mark: |
+| frontend                                   | manual (testing) | :x:                 | :x:             | :x:                             | :x:                |
+
+Steps executed if _Checks_ is checked :heavy_check_mark::
+
+### General
+- Check CircleCi config
+
+### Administration
+- Lint Administration
+- GraphQL scheme
+- Prettier formatting
+- TypeScript checks
+- Unit testing with jest
+
+### Backend
+- KLint
+- Integration & Unit tests
+- GraphQL scheme
+- Backend health
+
+### Frontend
+- Dart formatting
+- Build runner
+- Check Analyzer and linting
+- Unit tests
+
+For `commit_main` additionally all packages will be build and and backend health check will be done
+
+Steps executed if _Version bump_ is checked :heavy_check_mark::
+
+## Failed Delivery
+
+In case a delivery fails, please do not restart the workflow, but trigger a fresh release, to avoid another failure and version inconsistencies.
```

Comment:
Hmm.... I think this depends on the case of the failure. If there was only a network outage or it was rejected by apple bc an agreement was missing, I think it is better to restart from failed.

Reply:
added some additional text